### PR TITLE
fix: terraform plan parse bug

### PR DESCRIPTION
**Pull request by rontalx**
#### What does this PR do?
Adds meaningful error messages to Terraform plan scanning and fixes cases where the child modules array is missing from the data structure.
The bug was in the reducing function which assumed `child_modules` is always present and didn't have safe defaults.

### CODE REVIEW

File: src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts [hunk 1]
```diff
@@ -30,14 +30,14 @@ function terraformPlanReducer(
 function extractRootModuleResources(
   terraformPlanJson: TerraformPlanJson,
 ): Array<TerraformPlanResource> {
-  return terraformPlanJson?.planned_values?.root_module?.resources;
+  return terraformPlanJson?.planned_values?.root_module?.resources || [];
```
Comment (aron) [inline src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts:33]: Nit:
```suggestion
  return terraformPlanJson?.planned_values?.root_module?.resources ?? [];
```

**Review by p0tr3c — APPROVED** [review]

File: src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts [hunk 2]
```diff
@@ -47,23 +47,35 @@ function extractChildModulesResources(
...
-  const terraformPlanJson = JSON.parse(
-    terraformPlanFile.fileContent,
-  ) as TerraformPlanJson;
+  let terraformPlanJson;
+  try {
+    terraformPlanJson = JSON.parse(
+      terraformPlanFile.fileContent,
+    ) as TerraformPlanJson;
+  } catch (err) {
+    throw new Error('Failed to parse Terraform plan JSON file.');
```
Comment (aron) [inline src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts:56]: Likely not an issue now but `JSON.parse()` throws a SyntaxError so you can catch explicit parse errors and re-throw anything else that happens to be caught, like type/reference errors.

File: src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts [hunk 3]
```diff
@@ -47,23 +47,35 @@ function extractChildModulesResources(
...
+      {
+        ...terraformPlanFile,
+        jsonContent: parsedInput,
+        engineType: EngineType.Terraform,
+      },
+    ];
+  } catch (err) {
+    throw new Error(
+      'Failed to extract resources from Terraform plan JSON file.',
+    );
```
Comment (aron) [inline src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts:79]: Ditto here, we probably want to throw a specific Error subclass and detect here and then rethrow any other errors. Does the CLI have a logger we can run in debug mode to log the underlying error here to help users debug the issue?

**Review by aron — APPROVED** [review]
Some suggestions for future error handling inline, we'll likely want to get these in for beta.

File: src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts (hunk 3 above)
Comment (rontalx) [reply src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts:79]: agreed.
I want to include that work in the `CLI interface overhaul` ticket, to also map all of our potential errors and fine-tuning them with proper classes and messages.
not sure about the debug logger, worth checking.

**Pull request merged**